src/data/set/lattice.lean

```diff
@@ -358,6 +358,14 @@ theorem bUnion_union (s t : set α) (u : α → set β) :
   (⋃ x ∈ s ∪ t, u x) = (⋃ x ∈ s, u x) ∪ (⋃ x ∈ t, u x) :=
 supr_union
 
+@[simp] lemma bUnion_of_singleton_of_coe (s : set α) :
+  (⋃ (i : s), {i} : set α) = s :=
+ext $ by simp
+
+lemma set.Union_subtype {α β : Type*} (s : set α) (f : α → set β) :
+  (⋃ (i : s), f i) = ⋃ (i ∈ s), f i :=
+(set.bUnion_eq_Union s $ λ x _, f x).symm
```

Comment:
For other reviewers - is this a good `simp` lemma? My feeling is yes, because we have a lot of other simp lemmas about things of the form `⋃ (i ∈ s)` when `s` is some compound expression.